explorer/interpreter/interpreter.cpp

```diff
@@ -751,7 +779,9 @@ auto Interpreter::CallFunction(const CallExpression& call,
     }
     case Value::Kind::FunctionValue: {
       const FunctionValue& fun_val = cast<FunctionValue>(*fun);
-      const FunctionDeclaration& function = fun_val.declaration();
+      const CallableDeclaration& callableDeclaration = fun_val.declaration();
+      const FunctionDeclaration* function =
+          dyn_cast<FunctionDeclaration>(&callableDeclaration);
```

Comment:
For FunctionValue and BoundMethodValue, it looks like the code assumes they have a FunctionDeclaration. Should they be switched back to storing a FunctionDeclaration rather than a CallableDeclaration?

Also, note `callableDeclaration` should be `callable_declaration` here, but I wanted to focus on the type.

Reply:
Yes!

Reply:
Done!